src/test/java/uk/gov/companieshouse/acsp/manage/users/common/ComparisonUtils.java

```diff
@@ -65,4 +67,19 @@ public <T> ArgumentMatcher<T> compare( final T referenceObject, final List<Strin
         };
     }
 
+    public static ArgumentMatcher<Update> updateMatches( final Map<String, Object> expectedKeyValuePairs ) {
```

Comment:
where is this being used?

Reply:
There are three tests in AcspMembersServiceTest that are using it. I considered leaving it in that file because its the only place its used, but I thought that the method is general enough that it might be usable elsewhere, if we add extra tests later. These are the tests:


```
    @Test
    void updateMembershipWithNullUserStatusAndNotNullUserRoleOnlyUpdatesEtagAndRole() {
        Mockito.doReturn( 1 ).when( acspMembersRepository ).updateAcspMembership( eq( "TS001" ), any( Update.class ) );
        acspMembersService.updateMembership("TS001", null, UserRoleEnum.STANDARD, "TSU002" );
        Mockito.verify(acspMembersRepository).updateAcspMembership( eq("TS001"), argThat( updateMatches( Map.of("user_role", UserRoleEnum.STANDARD.getValue() ) ) ) );
    }

    @Test
    void updateMembershipWithNotNullUserStatusAndNullUserRoleOnlyUpdatesEtagAndStatusAndRemovedAtAndRemovedBy() {
        Mockito.doReturn( 1 ).when( acspMembersRepository ).updateAcspMembership( eq( "TS001" ), any( Update.class ) );
        acspMembersService.updateMembership( "TS001", UserStatusEnum.REMOVED, null, "TSU002" );
        Mockito.verify( acspMembersRepository ).updateAcspMembership(eq( "TS001" ), argThat( updateMatches( Map.of("status", UserStatusEnum.REMOVED.getValue(), "removed_by", "TSU002") ) ) );
    }

    @Test
    void updateMembershipWithNotNullUserStatusAndNotNullUserRoleOnlyUpdatesEverything() {
        Mockito.doReturn( 1 ).when( acspMembersRepository ).updateAcspMembership( eq( "TS001" ), any( Update.class ) );
        acspMembersService.updateMembership( "TS001", UserStatusEnum.REMOVED, UserRoleEnum.STANDARD, "TSU002" );
        Mockito.verify( acspMembersRepository ).updateAcspMembership( eq( "TS001" ), argThat( updateMatches( Map.of("user_role", UserRoleEnum.STANDARD.getValue(), "status", UserStatusEnum.REMOVED.getValue(), "removed_by", "TSU002" ) ) ) );
    }
```